sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -0,0 +1,971 @@
+    @SuppressWarnings("deprecation")
+    private static Uri getFileUriOnOldDevices(Cursor cursor) {
+        return Uri.parse("file://" + cursor.getString(cursor.getColumnIndex(DownloadManager.COLUMN_LOCAL_FILENAME)));
+    }
+
+    @SuppressWarnings("deprecation")
+    private static Notification buildNotification(Notification.Builder builder) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            return builder.build();
+        } else {
+            return builder.getNotification();
+        }
+    }
+
+    @Override
+    protected String getGroupName() {
+        return null;
+    }
+
+    @Override
+    protected String getServiceName() {
+        return SERVICE_NAME;
+    }
+
+    @Override
+    protected String getLoggerTag() {
+        return LOG_TAG;
+    }
+
+    @Override
+    public synchronized void onStarted(@NonNull Context context, @NonNull String appSecret, @NonNull Channel channel) {
+        super.onStarted(context, appSecret, channel);
+        mContext = context;
+        mAppSecret = appSecret;
+        resumeUpdateWorkflow();
+    }
+
+    @Override
+    public synchronized void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+
+        /* Resolve launcher class name only once, use empty string to cache a failed resolution. */
+        if (mLauncherActivityClassName == null) {
+            mLauncherActivityClassName = "";
+            PackageManager packageManager = activity.getPackageManager();
+            Intent intent = packageManager.getLaunchIntentForPackage(activity.getPackageName());
+            if (intent != null) {
+                mLauncherActivityClassName = intent.resolveActivity(packageManager).getClassName();
+            }
+        }
+
+        /* Clear workflow finished state if launch recreated, to achieve check on "startup". */
+        if (activity.getClass().getName().equals(mLauncherActivityClassName)) {
+            MobileCenterLog.info(LOG_TAG, "Launcher activity restarted.");
+            if (StorageHelper.PreferencesStorage.getString(PREFERENCE_KEY_DOWNLOAD_URI) == null) {
+                mWorkflowCompleted = false;
+                mBrowserOpened = false;
+            }
+        }
+    }
+
+    @Override
+    public synchronized void onActivityResumed(Activity activity) {
+        mForegroundActivity = activity;
+        resumeUpdateWorkflow();
+    }
+
+    @Override
+    public synchronized void onActivityPaused(Activity activity) {
+        mForegroundActivity = null;
+    }
+
+    @Override
+    public synchronized void setInstanceEnabled(boolean enabled) {
+        super.setInstanceEnabled(enabled);
+        if (enabled) {
+            resumeUpdateWorkflow();
+        } else {
+
+            /* Clean all state on disabling, cancel everything. */
+            mBrowserOpened = false;
+            mWorkflowCompleted = false;
+            cancelPreviousTasks();
+            StorageHelper.PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_TOKEN);
```

Comment:
Disable forgets about the state like every beacon does.
The improvement I would suggest if it matters is to wait process or launcher restart to restart a new workflow.